### ISSUE TITLE
Issue in IDF file when OSM file has thermal zone with same name as space within it

### ISSUE THREAD

**Issue by DavidGoldwasser** [NREL/OpenStudio#1618]
In IDF thermal zone ends up with handle as name instead of the OS:ThermalZone name.
https://unmethours.com/question/4503/zone-handle-instead-of-name-e-file-output-not-os/?answer=4513#post-id-4513

**Comment by macumber** [comment]
Did you reproduce this @DavidGoldwasser?  Can we give this one to @kbenne I'm feeling a little overloaded on bugs lately

**Comment by DavidGoldwasser** [comment]
No, I haven't tried to reproduce it. @kbenne if you want to assign to me first I can reproduce it to confirm it happens, but the user description was pretty detailed.

**Comment by DavidGoldwasser** [comment]
I wasn't able to break it with some monkey testing. I did multiple spaces in zone with different space types. I matched zone and space name. GUI prevented me from also using duplicate name of space/zone for the space type. At first glance it appears like zone list names come from the space type name.

**Comment by DavidGoldwasser** [comment]
User sent a file, and I was able to reproduce the issue.Somehow the user was able to make SpaceType names matching thermal zones names. (I must have doing something wrong before, or things have changed after 1.7.0. In 1.7.0 I can reproduce matching spaceType/thermal zone names.
https://github.com/NREL/OpenStudio-files/tree/master/Git%20Issue%20Files/Issue1618

In most cases the zone list takes on what was the zone/spaceType name, and the zone is a guid.
```
ZoneList,
  Bag Dump,                               !- Name
  {5e02f9fb-ea5a-464b-a140-34f07c41da8d}; !- Zone Name 1
```

In one case the zone kept its name and the zone list got guid
```
  {0be6dafa-9a8d-45f9-a3db-40a7d6c9aacf}, !- Name
  Blower Room and Sanitation Office;      !- Zone Name 1
```

When space type names have prefix added, no guid's are needed for names.
```
ZoneList,
  ST: Cooler,                             !- Name
  Cooler;                                 !- Zone Name 1
```

**Comment by DavidGoldwasser** [comment]
@macumber seems like one for you. While this seems like may be minor, it is creating issue with some reporting measures. I'll look into specifically the annual end use one. It may be expecting zone names from the OSM to be in SQL results? not sure. Standard report does seem to run.

```
The attached model is the subject of two recent unmet hours threads:
"Zone handle instead of name (E+ file & Output not OS)"
"Openstudio run fails after successful simulation"

For the naming issue, you may simulate this model and view its SQL output and see that one of the names is a label.  If you go to space types and retype "ST: " in front of the first few (I removed it for this
test) it will go back to being normal.  It seems in this model that the space type name cannot equal the zone name for some cases.  See the thread for a detailed discussion.

For the other problem, if you add many of the QAQC measures, the OS run will fail after a successful E+ simulation.  The measure I am interested in is: annual_end_use_breakdown, while I tried a few other measures last night that also failed the model.  A fix would be good but I'd also like to know the first steps for local debugging - e.g. is there an OS.err equivalent?  Note, the current model is set up for 1month the failures were noticed after annual simulation.

A fix for the second issue is more preferred as the renaming in the first issue is just a convenience, not a functionality limitation.
```

**Comment by DavidGoldwasser** [comment]
Ok, so when I tried this last week, I tried to name the space, space type, and thermal zone all with the same name. That isn't allowed. 

- There is protection against a space and a space type having the same name
- There is protection against a space and and thermal zone having the same name.

- There is not any protection against a space type and a thermal zoen having the same name. That seems to be the cause of this issue. I tested this be renaming the space to I wasn't trying all three at once. Maybe we just lock this done in GUI and or API. I assume API is preferable if it doesn't create any issues.

**Comment by DavidGoldwasser** [comment]
FYI - the reporting measure error he was seeing was un-related to the thermal zone names. It was a typo in the measure which I'm fixing now.
https://unmethours.com/question/4715/openstudio-run-fails-after-successful-simulation/?answer=4759#post-id-4759

**Comment by MatthewSteen** [comment]
@DavidGoldwasser regarding your 3 bullets above, I have routinely violated #2 as part of my regular workflow without problem. In SU and text editor:

1. Add New Thermal Zone for Spaces With No Thermal Zone
2. Rename Thermal Zones Based on Space Names
3. Open OSM in text editor and strip "Thermal Zone: " from all OS:ThermalZone object names




and have never run into a problem. he same name for spaces and thermal zones in my models. I use the built in SU user script and then strip the "Thermal Zone: " prefix so that the spaces
